Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2003,19 +2044,39 @@ namespace Babylon
 
                     current.insert(inputSource.ID);
 
-                    auto found = m_idToInputSource.find(inputSource.ID);
-                    if (found == m_idToInputSource.end())
+                    auto inputSourceFound = m_idToInputSource.find(inputSource.ID);
+                    if (inputSourceFound == m_idToInputSource.end())
                     {
                         // Create the new input source, which will have the correct spaces associated with it.
                         m_idToInputSource.insert({inputSource.ID, CreateXRInputSource(inputSource, env)});
+                        
+                        //Now that input Source is created, create a gamepad object if enabled for the input source
+                        inputSourceFound = m_idToInputSource.find(inputSource.ID);
```

Comment:
I agree, the reason i did this is because when I tried to save the output of CreateXRInputSource as a Napi::ObjectReference and use it later for gamepad creation, the m_idToInputSource::insert is not allowing to insert the saved napi::objectReference for some overloading reasons, not sure why.